src/runtime-rs/crates/hypervisor/ch-config/Cargo.toml

```diff
@@ -0,0 +1,18 @@
+[package]
+name = "ch-config"
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+anyhow = "1.0.68"
+serde = { version = "1.0.145", features = ["rc", "derive"] }
+serde_json = "1.0.91"
+tokio = { version = "1.25.0", features = ["sync", "rt"] }
+
+# Cloud Hypervisor public HTTP API functions
+# Note that the version specified is not necessarily the version of CH
+# being used. This version is used to pin the CH config structure
+# which is relatively static.
```

Comment:
Is the CH client API stable? I wonder if it can still work if a different CH version is used.

Reply:
My understanding is that it's relatively stable, yes. The pinning is for safety: Although Cloud Hypervisor provides a set of public client API calls, it doesn't provide a public definition of its internal configuration type (`VmConfig`), at least, not in a form we can use [*], so this PR (has to) contain "an implementation" of the _current_ `VmConfig` type. This approach was adopted at the recommendation of the Cloud Hypervisor developers.

So, it _might_ work if we don't pin, but there are no guarantees :cry: 

---

[*] - CH does provide a [YAML format OpenAPI spec file](https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/vmm/src/api/openapi/cloud-hypervisor.yaml), but the tooling to convert this into rust is not reliable. Plus, there is some additional complexity around the differences between OpenAPI defined data types defined versus rust lang data types. See https://github.com/cloud-hypervisor/cloud-hypervisor/issues/4732 for further details.